python/social_bot/teacher_tasks.py

```diff
@@ -580,31 +579,23 @@ def __init__(self,
             random_range (float): the goal's random position range
             target_speed (float): the target speed runing to the ball. The agent will receive no more 
                 higher reward when its speed is higher than target_speed.
-            step_time (float): used to caculate speed of the agent
             reward_weight (float): the weight of the reward
         """
-        GoalTask.__init__(
-            self,
+        super().__init__(
             max_steps=max_steps,
             goal_name=goal_name,
             fail_distance_thresh=fail_distance_thresh,
             random_range=random_range)
         self._goal_name = 'goal'
-        self._success_distance_thresh = success_distance_thresh,
-        self._objects_in_world = [
-            'placing_table', 'plastic_cup_on_table', 'coke_can_on_table',
-            'hammer_on_table', 'cafe_table', 'ball'
-        ]
-        self._step_time = step_time
+        self._success_distance_thresh = success_distance_thresh
         self._target_speed = target_speed
         self.reward_weight = reward_weight
```

Comment:
pass reward_weight to super(), instead of having a separate variable in this derived class.